library/src/test/java/com/novoda/downloadmanager/lib/DownloadStatusTest.java

```diff
@@ -0,0 +1,127 @@
+package com.novoda.downloadmanager.lib;
+
+import android.app.DownloadManager;
+
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+public class DownloadStatusTest {
+
+    @Test
+    public void whenVisibilityIsCompletedThenNotificationIsToBeDisplayed() {
+        boolean toBeDisplayed = DownloadStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
+        assertThat(toBeDisplayed).isTrue();
+    }
+
+    @Test
+    public void whenVisibilityIsOnlyCompletionThenNotificationIsToBeDisplayed() {
+        boolean toBeDisplayed = DownloadStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_ONLY_COMPLETION);
+        assertThat(toBeDisplayed).isTrue();
+    }
+
+    @Test
+    public void whenVisibilityIsHiddenThenNotificationIsNotBeDisplayed() {
+        boolean toBeDisplayed = DownloadStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_HIDDEN);
+        assertThat(toBeDisplayed).isFalse();
+    }
+
+    @Test
+    public void whenStatusIs490ThenStatusIsCancelled() {
+        boolean isCancelled = DownloadStatus.isCancelled(490);
+        assertThat(isCancelled).isTrue();
+    }
+
+    @Test
+    public void whenStatusIs4xxThenStatusIsClientError() {
```

Comment:
these tests should be only contain one assertion IMO

or at least one logic assertion (i.e. extract the for-loop in a separate method?) or just have 3 tests:
 - one for 399
 - one for 450
 - one for 500

is it necessary to test all the values between 400 and 500? then why don't you test all the rest :tongue: 

would be nice to see more concise tests http://programmaticallyspeaking.com/one-assertion-per-test-please.html